includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -571,85 +571,71 @@ public function is_page_supported() {
+				break;
+			case WC_Stripe_Helper::has_cart_or_checkout_on_current_page() && ! $this->allowed_items_in_cart():
+				// Don't show if on the cart or checkout page, or if page contains the cart or checkout
+				// shortcodes, with items in the cart that aren't supported.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due some items in cart not being not compatible with it. ' );
+				break;
+			case is_cart() && ! $this->should_show_ece_on_cart_page():
+				// Don't show on cart if disabled.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due buttons display on cart being disabled. ' );
+				break;
+			case is_checkout() && ! $this->should_show_ece_on_checkout_page():
+				// Don't show on checkout if disabled.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due buttons display on checkout being disabled. ' );
+				break;
+			case $this->is_product() && ! $this->should_show_ece_on_product_pages():
+				// Don't show if product page ECE is disabled.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due buttons display on product pages being disabled. ' );
+				break;
+			case $this->is_product() && ! $this->is_product_supported( $product ):
+				// Don't show if product on current page is not supported.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due product not being supported by Stripe Express Checkout. Product ID: ' . $product->get_id() );
+				break;
+			case $this->is_product() && ! in_array( true, $stock_availability, true ):
+				// Don't show if all product variations are out-of-stock.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due to product variations being out of stock. Product ID: ' . $product->get_id() );
+				break;
+			case ( ! $this->is_pay_for_order_page() && ! $this->is_product() && $is_free_cart )
+				|| ( $this->is_product() && (float) $this->get_product()->get_price() === 0.0 ):
+				// Don't show if the total price is 0.
+				// ToDo: support free trials. Free trials should be supported if the product does not require shipping.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due to cart amount or product price being 0. ' . print_r( [ 'url' => get_permalink() ], true ) );
+				break;
+			case $this->should_hide_ece_based_on_tax_setup():
+				// Hide if cart/product doesn't require shipping and tax is based on billing or shipping address.
+				WC_Stripe_Logger::log( 'Stripe Express Checkout may be hidden due to product/cart not requiring shipping and tax being based on customer\'s billing or shipping address.' );
+				break;
+			default:
+				// If all checks pass, show the button by default.
+				$should_show_button = true;
 		}
 
-		return true;
+		// Allow other plugins to filter the visibility of the button.
+		return apply_filters( 'wc_stripe_should_show_express_checkout_button', $should_show_button );
```

Comment:
Optional: If `apply_filters()` returns a value that is not the same as `$should_show_button` it may be useful to log the override.

Reply:
Added the log in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4200/commits/922836a87c6239ec8bcdc3ea167664ec234132dc